docs/Streams.md

```diff
@@ -74,38 +74,100 @@ If a stream gets canceled because it is in 'cancel on loss' mode, a `QUIC_STREAM
\ No newline at end of file
+Multi-receive mode is a connection wide option allowing multiple receive notification to be pending simultaneously.
+It is enabled by setting [`StreamMultiReceiveEnabled`](./Settings.md) in connection parameters.
+
+For streams created when the connection is in Multi-mode receive, MsQuic can keep indicating `QUIC_STREAM_EVENT_RECEIVE` before the application completes the previous one.
+This means that the application must be able to handle a new `QUIC_STREAM_EVENT_RECEIVE` even if it returned `QUIC_STATUS_PENDING` previously and has not called [`StreamReceiveComplete`](api/StreamReceiveComplete.md) yet.
+
+MsQuic will also keep indicating receive notifications when the application accept the data partially. The bytes that have not been accepted by the application won't be indicated again: the application must call [StreamReceiveComplete](api/StreamReceiveComplete.md) in the future to accept them.
+
+To handle multi-receive mode properly, the application must keep of the total number of byte received on the stream (the sum of all `TotalBufferLength`).
+The number of calls to [`StreamReceiveComplete`](api/StreamReceiveComplete.md) does not need to be equal to the number of receive notification, but the total number of bytes received must eventually be equal to the total number of bytes accepted.
+
+Multi-receive mode manages its internal receive buffer differently and is more efficient for continuous receiving with asynchronous processing.
+
+### App-owned buffer mode
+
+App-owned buffer mode is a per-stream option allowing the application to provide its own receive memory buffers.
+Enabling app-owned mode is done differently depending on whether the stream is created locally or from the peer and is discussed below.
+
+When in app-owned mode, the application can call [`StreamProvideReceiveBuffers`](./????.md) to provide a list of memory buffers to MsQuic.
+[`StreamProvideReceiveBuffers`](./????.md) can be called at any time on a valid stream in app-owned mode, potentially inline from a notification handler.
+If called several times, the buffer provided through subsequent calls are added to the list.
+
+MsQuic will fill the provided buffer with received data, in the order they have been provided. Receive notification will be emitted as normal,
+indicating a list of `QUIC_BUFFER` pointing to the application provided buffers.
+Note that up to the number of buffers the application provided can be indicated at once, and that only part of a buffer can be indicated.
+There is no guarantees the `QUIC_BUFFER`s indicated in a receive notification will match the ones the application provided.
+
+The application is responsible for tracking the amount of data received and when a buffer it provided has been fully used.
+The application regains full ownership of a buffer after it get a receive notification for all bytes in the buffer and accept them by calling [`StreamMultiReceiveEnabled`](./Settings.md).
```

Comment:
```suggestion
The application regains full ownership of a buffer after it get a receive notification for all bytes in the buffer and accepts them by calling [`StreamReceiveComplete`](./Settings.md).
```
It's receive complete, not receive enabled? Also, please fix up all these links that point to `Settings.md`

Reply:
Yes, that looks like a copy-past gone wrong.